src/cmd/singularity/actions_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+	"testing"
+
+	"github.com/sylabs/singularity/src/pkg/test"
+)
+
+//build base image for tests
+const imagePath = "./container.img"
+
+type execOpts struct {
+	binds   []string
+	contain bool
+	home    string
+	workdir string
+	pwd     string
+}
+
+func imageExec(opts execOpts, imagePath string, command []string) ([]byte, error) {
+	argv := []string{"exec"}
+	for _, bind := range opts.binds {
+		argv = append(argv, "--bind", bind)
+	}
+	if opts.contain {
+		argv = append(argv, "--contain")
+	}
+	if opts.home != "" {
+		argv = append(argv, "--home", opts.home)
+	}
+	if opts.workdir != "" {
+		argv = append(argv, "--workdir", opts.workdir)
+	}
+	if opts.pwd != "" {
+		argv = append(argv, "--pwd", opts.pwd)
+	}
+	argv = append(argv, imagePath)
+	argv = append(argv, command...)
+
+	return exec.Command(cmdPath, argv...).CombinedOutput()
+}
+
+// testSingularityRun tests min fuctionality for singularity run
+func testSingularityRun(t *testing.T) {
+	tests := []struct {
+		binName string
+		name    string
+		argv    []string
+		exit    int
+	}{
+		{cmdPath, "NoCommand", []string{"run", imagePath}, 0},
+		{cmdPath, "true", []string{"run", imagePath, "true"}, 0},
+		{cmdPath, "false", []string{"run", imagePath, "false"}, 1},
+		// Testing run command properly hands arguments
+		{"sh", "arguments", []string{"-c", fmt.Sprintf("singularity run %s foo | grep foo", imagePath)}, 0},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, test.WithoutPrivilege(func(t *testing.T) {
+			cmd := exec.Command(tt.binName, tt.argv...)
```

Comment:
If we've created a function to run an image, then why are we not using it?

Reply:
`imageExec()` does not catch and return the exit code. should we modify that func to do it. or re allocate `imageExec()` to a diff test file